src/Krank.hs

```diff
@@ -1,13 +1,33 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
 module Krank (
-  processFile
+  runKrank
   ) where
 
 import qualified Krank.Checkers.IssueTracker as IT
 import Krank.Types
 import Control.Monad.Reader
+import Control.Exception.Safe
+import PyF
+import System.IO (hPutStrLn, stderr)
+import Data.Text (unpack)
+
+import Krank.Formatter
 
 processFile :: FilePath      -- ^ the file to analyze
-            -> ReaderT KrankConfig IO [Violation]
+            -> ReaderT KrankConfig IO ()
 processFile filePath = do
   content <- liftIO $ readFile filePath
-  IT.checkText filePath content
+
+  violations <- IT.checkText filePath content
+
+  KrankConfig{useColors} <- ask
+
+  liftIO $ putStr . unpack . foldMap (showViolation useColors) $ violations
```

Comment:
nit: any reason to jump a line each time?

Reply:
No. I can compact a bit.